config.example.wikimedia.yaml

```diff
@@ -1,265 +1,68 @@
-              # (TODO: remove after actual purging is implemented
-              #purge:
-              #  host: '239.128.0.112'
-              #  port: 4827
-              # For local testing, use:
-              # parsoidHost: http://localhost:8000
-
-      /{module:mobileapps}:
-        x-subspec:
-          paths:
-            /v1:
-              x-modules:
-                - name: mobileapps-sys
-                  path: mods/wikimedia/v1/mobileapps
-                  type: spec
-                  options:
-                    host: http://appservice.wmflabs.org
-
-  global-content: &gb/content/1.0.0
-    swagger: '2.0'
-    info: *wp/content-info/1.0.0
-    securityDefinitions: *wp/content-security/1.0.0
-    # Override the base path for host-based (proxied) requests. In our case,
-    # we proxy https://{domain}/api/rest_v1/ to the API.
-    x-host-basePath: /api/rest_v1
-    x-modules:
-      - name: mathoid
-        path: specs/media/v1/mathoid
-        type: spec
-        options:
-          host: http://mathoid-tester.wmflabs.org
-    paths:
-      /metrics:
-        x-subspecs:
-          - analytics/v1/pageviews
-
-  global-sys: &gb/sys/1.0.0
-    info:
-      title: global domain sys API module
-      version: 1.0.0
-    paths:
-      /{module:table}: *wp/sys/table
-      /{module:key_value}: *wp/sys/key_value
-      /{module:post_data}: *wp/sys/post_data
-
-  wp-default-1.0.0: &wp/default/1.0.0
-    x-subspecs:
-      - paths:
-          /{api:v1}:
-            x-subspec: *wp/content/1.0.0
-      - paths:
-          /{api:sys}:
-            x-subspec: *wp/sys/1.0.0
-
-# Should be used instead of wp-default-1.0.0 for private wiki domains
-  wp-secure-1.0.0: &wp/secure/1.0.0
-    x-subspecs:
-      - paths:
-          /{api:v1}:
-            x-subspec: *wp/content/1.0.0
-            security:
-              - mediawiki_auth:
-                - read
-      - paths:
-          /{api:sys}:
-            x-subspec: *wp/sys/1.0.0
-
-  global-default-1.0.0: &gb/default/1.0.0
-    x-subspecs:
-      - paths:
-          /{api:v1}:
-            x-subspec: *gb/content/1.0.0
-      - paths:
-          /{api:sys}:
-            x-subspec: *gb/sys/1.0.0
-
-
-spec: &spec
+# RESTBase wikimedia example config
+
+# Load some project templates. These are referenced / shared between domains
+# in the root_spec further down.
+default_project: &default_project
+  x-modules:
+    /:
+      - path: projects/wmf_default.yaml
+        options: &default_options
+          table:
+            hosts: [localhost]
+            keyspace: system
+            username: cassandra
+            password: cassandra
+            defaultConsistency: one # or 'localQuorum' for production
+            storage_groups:
+              - name: default.group.local
+                domains: /./
+          parsoid:
+            host: http://parsoid-lb.eqiad.wikimedia.org
+          action:
+            apiUriTemplate: "{{'https://{domain}/w/api.php'}}"
```

Comment:
@Pchelolo: To me the "hard to reason about this" bit is crucial. In order to support this, we'd have to change how templates are being expanded to ignore any errors. That would cost us performance and (more importantly) the ability to notice errors quickly.